### ISSUE TITLE
Support symlinks

### ISSUE THREAD

**Issue by qwerty287** [fclairamb/ftpserver#980]
Hello,
ftpserver does not support symlinks yet, they're just ignored. It would be very nice if it would be supported! Thank you!

**Comment by fclairamb** [comment]
Yes, very true. This needs to be implemented by adding [this API](https://github.com/fclairamb/ftpserverlib/blob/76e3b6785cda1e30aea2e19e78a0919034a76d24/driver.go#L83-L90).

**Comment by fclairamb** [comment]
@claude do you see how that could be done ?

**Comment by martinlindhe** [comment]
Assuming Linux, here's some solutions that works today:

A) Run ftpserver in a container and mount volumes as you wish. 

B) Use bind mounts. Requires you to edit /etc/fstab

C) Use bindfs. Requires no changes to fstab.

I use strategy A to make available a NAS share mounted on the host inside the ftpserver container, so i can reach it from a Windows XP vm. Works great!

**Comment by fclairamb** [comment]
Implemented in #1719. `ClientDriver` now supports the `SITE SYMLINK` command via ftpserverlib's `ClientDriverExtensionSymlink` interface, delegating to the underlying filesystem when it can create symlinks:

- the local **OS backend** creates real symlinks;
- backends that don't support them (S3, Dropbox, GDrive, …) return a "symlink not supported" error rather than failing opaquely.

This covers symlink *creation*; reading link targets (`Readlink`) isn't wired up in ftpserverlib yet. For the "expose host paths" use case, the container/bind-mount approaches @martinlindhe described above remain good options too.
